src/frequenz/dispatch/_actor_dispatcher.py

```diff
@@ -337,3 +327,13 @@ async def _handle_dispatch(self, dispatch: Dispatch) -> None:
             await self._start_actor(dispatch)
         else:
             await self._stop_actor(dispatch, "Dispatch stopped")
+
+
+def _convert_target_components(target: ClientTargetComponents) -> TargetComponents:
+    if all(isinstance(comp, int) for comp in target):
+        # We've confirmed all elements are integers, so we can cast.
+        int_components = cast(list[int], target)
+        return [ComponentId(cid) for cid in int_components]
+    # If not all are ints, then it must be a list of ComponentCategory
+    # based on the definition of ClientTargetComponents.
+    return cast(list[ComponentCategory], target)
```

Comment:
I don ´t get this, wasn't the whole point of the new target classes to enable using `match` more easily? Why do you have TargetComponents = list[ComponentId] | list[ComponentCategory]` here which makes distinguishing between the different ways to set targets harder? Shouldn't you just use the same `TargetComponents` defined by the client here so end users can just do:

```py
match target:
    case TargetIds() as ids:
        ...
    case TargetICategories() as cats:
        ...
```
?

If you really need to keep the local `TargetComponents`, can't this function be implemented using `match`?

```py
match target:
    case TargetIds() as ids:
        return list(ids)
    case TargetICategories() as cats:
        return list(cats)
```